test/unit/overlay.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+var fixtures = require('../fixtures');
+var fs = require('fs');
+var gm = require('gm');
+var sharp = require('../../index');
+
+sharp.cache(0);
+
+
+// Helpers
+var tryDelete = function (file) {
+  try {
+    fs.unlinkSync(file);
```

Comment:
Thanks for `fixtures.assertSimilar` and your example use of `toBuffer`. This enabled me to kill the intermediate files altogether :smile: